scripts/core/editor3/reducers/editor3.tsx

```diff
@@ -89,18 +93,116 @@ export const onChange = (state, newState, force = false) => {
     }
 
     if (force) {
-        return forceUpdate({
-            ...state,
-            editorState,
-        });
+        return forceUpdate(
+            applyAbbreviations({
+                ...state,
+                editorState,
+            })
+        );
     }
 
-    return {
+    return applyAbbreviations({
         ...state,
         editorState,
-    };
+    });
 };
 
+/**
+ * @ngdoc method
+ * @name setAbbreviations
+ * @param {Object} state
+ * @param {Object} abbreviations
+ * @return {Object} returns new state
+ * @description Set the abbreviations dictionary
+ */
+const setAbbreviations = (state, abbreviations) => ({
+    ...state,
+    abbreviations,
+});
+
+/**
+ * @ngdoc method
+ * @name applyAbbreviations
+ * @param {Object} editorState
+ * @param {Array} abbreviations
+ * @return {Object} returns new state
+ * @description Handle the editor tab key pressed event
+ */
+const applyAbbreviations = (state) => {
+    const {editorState, abbreviations} = state;
+    const selection = editorState.getSelection();
+
+    if (!selection.isCollapsed() || abbreviations == null || Object.keys(abbreviations).length === 0) {
+        return state;
+    }
+
+    const content = editorState.getCurrentContent();
+    const block = content.getBlockForKey(selection.getStartKey());
+    const word = getAbbreviationText(block, selection.getStartOffset());
+    const keys = Object.keys(abbreviations);
+    const pattern = '\\b(' + keys.map((item) => escapeRegExp(item)).join('|') + ')(\\*)';
+    const found = word.text.match(new RegExp(pattern, 'g'));
+
+    if (found) {
+        const abbreviation = found[0].replace('*', '');
+        const newWord = abbreviations[abbreviation];
+
+        return replaceWord(state, {word, newWord}, true);
+    }
+
+    return state;
+};
+
+/**
+ * @ngdoc method
+ * @name getAbbreviationText
+ * @param {Object} block
+ * @param {Integer} offset
+ * @return {String} returns text that can contain abbreviation
+ * @description If no current selection from current position extract text that is
```

Comment:
> If no current selection from current position

Can you clarify this comment? Do you mean if the selection doesn't exist? Is this `draft-js` or browser-selection?

Reply:
It is about draft-js selection, but looks that this check is not present in code. I will add it.